src/Microsoft.ML.StandardLearners/Standard/Online/OnlineLinear.cs

```diff
@@ -253,10 +253,19 @@ private protected sealed override TModel TrainModelCore(TrainContext context)
         {
             Host.CheckValue(context, nameof(context));
             var initPredictor = context.InitialPredictor;
-            var initLinearPred = initPredictor as LinearModelParameters ?? (initPredictor as CalibratedPredictorBase)?.SubPredictor as LinearModelParameters;
-            Host.CheckParam(initPredictor == null || initLinearPred != null, nameof(context), "Not a linear predictor.");
-            var data = context.TrainingSet;
 
+            if (initPredictor is LinearModelParameters initLinearPred)
+                initLinearPred = (LinearModelParameters)initPredictor;
+            else if (initPredictor is CalibratedPredictorBase<LinearBinaryModelParameters, PlattCalibrator> calibrated)
+                initLinearPred = calibrated.SubModelParameters;
+            else
+            {
+                initLinearPred = null;
+                Host.CheckParam(initPredictor == null || initLinearPred != null, nameof(context),
```

Comment:
>Host.CheckParam(initPredictor == null || initLinearPred != null, [](start = 16, length = 64)

What I think I meant though was, you can just nuke this test and replace it with a throw exception, since if the two tests above *fail*, then these two will definitely be null. So there is no point in having this be a check.

Reply:
We can't drop it because, when initial predictor is not provided, we will always throw.

---
In reply to: [254933689](https://github.com/dotnet/machinelearning/pull/2435#discussion_r254933689) [](ancestors = 254933689)